### ISSUE TITLE
Cqrlog dies at startup:  munmap_chunk(): invalid pointer

### ISSUE THREAD

**Issue by OH1KH** [ok2cqr/cqrlog#530]
This is a weird thing.

It happens when I have selected my "ALL" log (abt 12500 qsos) and pressed "Open Log" in  Database connection form.
Just after first window opens Cqrlog dies.
It is not depending on open windows amount, I have several of them but it is just the same if I only have NewQSO opening.
Also GTK2 or QT& does not make difference although it feels line QT5 is more stabile.
From console startup without debug and with show rig communication it looks like this:

Not started rigctld process. (Run is set FALSE)
Waiting for rigctld @ 127.0.0.1:4532
Connected to rigctld
Sending: +\chk_vfo

munmap_chunk(): invalid pointer
Keskeytetty (luotiin core-tiedosto)  in english: interrupted, core dump file created 
[saku@hamtpad ~]$ 

I have several other (test) logs with less qsos and usually they open without problem. It is just this largest log that usually fails.

Additional information: 
I use external database for logs (Mariadb server localhost port 3306)
Free Pascal Compiler version 3.2.2 [2021/09/21] for x86_64
Lazarus-ide 2.0.12 SVN Revision: Unknown
ldd (GNU libc) 2.34

It looks like memory leak. WIth Lazarus debugger I can see:
![image](https://user-images.githubusercontent.com/12532164/179451659-4ea8e246-e578-4173-be36-07ec99a06db1.png)
![image](https://user-images.githubusercontent.com/12532164/179451890-4b52c805-a517-48ef-a597-a748b4ef0373.png)
![image](https://user-images.githubusercontent.com/12532164/179452007-8122d272-6bb1-4552-8edd-065073771f87.png)

It is always the same place, but my skills are not good enough to locate what line the break point is in pascal code.
(laz debugger should show last executed program line number to make bug catch easier)

**Comment by OH1KH** [comment]
I have still this error. And just with one, same, log. All others are starting OK.

While again trying to trace this bug I found out that when I try to open that log and get errors again and again then if I do Utils/configuration/import for the log before opening it it always opens after that!!

It even helps to make Utils/configuration/export before opening that log, but once it happened that it did not start after export. But after Import it has always worked, so far.
There are no differences in .ini file. It seems that export/import using the "Q" query and "trQ" transaction is the key function to release something (memory?)

But I did not yet get to the root cause. Why?

Some day I continue from this point....

**Comment by OH1KH** [comment]
To be continued...
I deleted the problem log again. Made sure that it was gone and did also Cqrlog restart few times to be sure log was not there any more.
Then created a new log with problem log number (003), imported Utils/configuration/import made same file that I was previously using with tests. Closed Cqrlog once again and restarted now opening the log and doing full adif import of 13k of qsos.
After doing that the log 003 has worked normally, same way as others.

While doing this I turned off database replicating and this could have been the key. It might have been that cross replicating did affect to database delete so that it was not fully deleted before I recreated it in first place.
